Firestore/core/src/firebase/firestore/nanopb/writer.cc

```diff
@@ -125,6 +123,30 @@ std::string StringWriter::Release() {
   return std::move(buffer_);
 }
 
+// GrpcByteBufferWriter
+
+namespace {
+
+bool AppendToGrpcBuffer(pb_ostream_t* stream,
+                        const pb_byte_t* buf,
+                        size_t count) {
+  auto buffer = static_cast<std::vector<grpc::Slice>*>(stream->state);
+  buffer->emplace_back(buf, count);
+  return true;
+}
+
+}  // namespace
+
+GrpcByteBufferWriter::GrpcByteBufferWriter() {
+  stream_.callback = AppendToGrpcBuffer;
+  stream_.state = &buffer_;
+  stream_.max_size = SIZE_MAX;
+}
+
+grpc::ByteBuffer GrpcByteBufferWriter::Release() {
+  return grpc::ByteBuffer{buffer_.data(), buffer_.size()};
```

Comment:
By virtue of implementation, this doesn't let go of ownership. Would it make sense to clear `buffer_` for consistency with the "default" `Writer`?

Reply:
`grpc_raw_byte_buffer_create` merely bumps the reference count of all the slices, so `Release` really should clear the underlying buffer in order to make the caller's use of the `ByteBuffer` deterministic. Otherwise the buffer's backing slices won't be destroyed until the writer is destroyed.